### ISSUE TITLE
PreloadJs not working.

### ISSUE THREAD

**Issue by mountain01** [ajbarry3/createjs-module#2]
When using preloadjs to load assets, no event is ever fired. Also when given a manifest to load, it seems to only ever load the first item in the manifest while never fetching the others.

**Comment by ajbarry3** [comment]
I've just updated the module to: createjs-module@0.8.2-24
Let me know if that works

**Comment by mountain01** [comment]
nope, still no luck. I created a new project just to test out the new version. Here's my app.component.ts. When I look at the network tab it still only shows that it grabbed the first item in the manifest.


``` javascript
import { Component, AfterViewInit } from '@angular/core';
import * as createjs from 'createjs-module';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent implements AfterViewInit{

  ngAfterViewInit(){
    const loader = new createjs.LoadQueue(false);
    loader.on('complete',()=>console.log('complete'));
    loader.on('fileload',()=>console.log('file loaded'));
    const manifest = [
      {src:'https://s-media-cache-ak0.pinimg.com/originals/58/94/06/5894069ef5a80d3861daf25e45c4eb71.png',id:'spaceship1'},
      {src:'http://img4.wikia.nocookie.net/__cb20121116190751/halo/images/8/87/Broadsword.png',id:'spcae2'},
    ];
    loader.loadManifest(manifest);

    const stage = new createjs.Stage('myCanvas');
    const ball = new createjs.Shape();
    stage.addChild(ball);
    ball.graphics.beginFill('red').drawCircle(0,0,50);
    ball.x = (stage.canvas as any).width/2;
    ball.y = (stage.canvas as any).height/2;

    createjs.Ticker.addEventListener('tick',stage);
  }

}
```

And my package.json
``` javascript
{
  "name": "preload",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^4.0.0",
    "@angular/common": "^4.0.0",
    "@angular/compiler": "^4.0.0",
    "@angular/core": "^4.0.0",
    "@angular/forms": "^4.0.0",
    "@angular/http": "^4.0.0",
    "@angular/platform-browser": "^4.0.0",
    "@angular/platform-browser-dynamic": "^4.0.0",
    "@angular/router": "^4.0.0",
    "core-js": "^2.4.1",
    "createjs-module": "^0.8.2-24",
    "rxjs": "^5.1.0",
    "zone.js": "^0.8.4"
  },
  "devDependencies": {
    "@angular/cli": "1.2.1",
    "@angular/compiler-cli": "^4.0.0",
    "@angular/language-service": "^4.0.0",
    "@types/jasmine": "~2.5.53",
    "@types/jasminewd2": "~2.0.2",
    "@types/node": "~6.0.60",
    "codelyzer": "~3.0.1",
    "jasmine-core": "~2.6.2",
    "jasmine-spec-reporter": "~4.1.0",
    "karma": "~1.7.0",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.1.2",
    "ts-node": "~3.0.4",
    "tslint": "~5.3.2",
    "typescript": "~2.3.3"
  }
}
```

**Comment by ajbarry3** [comment]
I've just updated the module to: createjs-module@0.8.2-27
The problem was with the types file i made. Let me know if this works so I can close the issue

**Comment by mountain01** [comment]
I just updated to 0.8.2-27 and unfortunately the behavior hasn't changed for me. It is still only loading the first entry in the manifest and the events I'm listening for aren't being broadcast.

**Comment by ajbarry3** [comment]
Try setting the LoadQueue to true as shown below
const loader = new createjs.LoadQueue(true);

**Comment by mountain01** [comment]
Still no luck unfortunately. When I put true instead of false I get a CORS error when trying to load the image, and it still only tries to get the first one.

**Comment by flackjap** [comment]
Hey, I had issues with CORS when trying to manipulate image positions (dragging on pressmove event) - and I am kinda sure that I did everything to ensure that the CORS rules are set properly and allow everything for the domain in use (I still get an error sometimes and I just refresh the page and everything works fine - I had to postpone solving that issue for some later time).

So, It may be that the problem you're having has nothing to do with PreloadJS, but with something deeper in CreateJS that is involved with CORS rules.

**Comment by ajbarry3** [comment]
The only other thing I can think of that could be causing the issue is your running on a localhost. Try uploading it to a server and see if that work. This issue is beyond my expertise. I suggest you open the issue with the developers. I'm just an amateur developer.

**Comment by mountain01** [comment]
Yeah, I'm not sure why it isn't working. Luckily it's not that big of a deal in my case. I was able to make my own very stripped down version of what PreloadJS does, but just for my specific use case. I've had a very positive experience with the rest of your module though, so thanks for putting it together.

**Comment by dpodsiadlo** [comment]
Hey guys! I've faced the same issue and I've found what's causing it. In PreloadS, there's a method 

     p._isCanceled = function () {
		if (window.createjs == null || this.canceled) {
			return true;
		}
		return false;
	};

which checks if there's a `window.createjs` variable declared. In my case it's undefined and the _isCanceled function always returns true, which stops loading queue every time any file is loaded. So it's definitely an issue with the original PreloadJS code. My quick fix for it was to declare the `window.createjs` variable like this:

    import * as createjs from 'createjs-module';
    (<any>window).createjs = createjs;
